io_ogre/ogre/material.py

```diff
@@ -175,11 +202,57 @@ def generate_pass( self, mat, pass_name="" ):
                         if node.texture:
                             geo = shader.get_connected_input_nodes( self.material, node )[0]
                             self.generate_texture_unit( node.texture, name=name, uv_layer=geo.uv_layer )
-            elif images:
-                for image in images:
-                    self.generate_texture_unit(image)
+            elif textures:
+                for key, texture in textures.items():
+                    self.generate_image_texture_unit(key, texture)
 
+    def generate_image_texture_unit(self, key, texture):
+        """
+        Generates a texture_unit of a pass.
+        
+        key: key of the texture in the material wrapper (not used, for normal map if needed)
+        texture: the texture wrapper (node_shader_utils.PrincipledBSDFWrapper)
+        """
+        src_dir = os.path.dirname(bpy.data.filepath)
+        # For target path relative
+        # dst_dir = os.path.dirname(self.target_path)
+        dst_dir = src_dir
+        filename = io_utils.path_reference(texture.image.filepath, src_dir, dst_dir, mode='RELATIVE', library=texture.image.library)
+        # Do not use if target path relative
+        # filename = repr(filepath)[1:-1]
+        _, filename = split(filename)
+        filename = self.change_ext(filename, texture.image)
```

Comment:
yes, we should keep this. It is used by `FORCE_IMAGE_FORMAT` which got lost in the 2.8 conversion